rspace-bench/src/main/scala/coop/rchain/rspace/bench/EvalBench.scala

```diff
@@ -0,0 +1,119 @@
+package coop.rchain.rspace.bench
+
+import java.io.{FileNotFoundException, InputStreamReader}
+import java.nio.file.{Files, Path}
+
+import cats.implicits._
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.models.Par
+import coop.rchain.rholang.interpreter.{Env, Interpreter, Runtime}
+import org.openjdk.jmh.annotations._
+import coop.rchain.shared.PathOps.RichPath
+import monix.eval.Task
+import monix.execution.{ExecutionModel, Scheduler}
+import monix.execution.schedulers.{CanBlock, TrampolineScheduler}
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+//for debug/check run from IDEA with
+//rspaceBench/jmh:run EvalBench.replayMvcEppST -i 1 -wi 0 -f 0 -t 1
+//on hyper-threaded machines for standalone check use -f <number of physical cores>
+//for example
+//java -jar target/scala-2.12/rspacebench_2.12-0.1.0-SNAPSHOT.jar EvalBench -i 10 -wi 5 -f 2 -t 2
+class EvalBench {
+  import EvalBench._
+
+  def processErrors(errors: Vector[Throwable]): Unit = if (errors.nonEmpty) {
+    throw new RuntimeException(
+      errors
+        .map(_.toString())
+        .mkString("Errors received during evaluation:\n", "\n", "\n"))
+  }
+
+  def prepareRunTask(state: MvcEppBenchState): Task[Vector[Throwable]] = {
+    val par = state.term.getOrElse(throw new Error("Failed to prepare executable rholand term"))
+    state.runtime.reducer
+      .inj(par)(state.rand)
+      .map(_ => state.runtime.readAndClearErrorVector())
+  }
+
+  //if we run multiple tests on a single-threaded scheduler
+  //they will compete on single execution queue
+  //therefore this test always limited to one thread
+  @Benchmark @Threads(1)
+  def replayMvcEppST(state: MvcEppBenchState): Unit = {
+    val runTask = prepareRunTask(state).executeOn(state.singleThreadedScheduler, forceAsync = false)
+    processErrors(
+      runTask.runSyncUnsafe(Duration.Inf)(state.singleThreadedScheduler, CanBlock.permit))
+  }
+
+  @Benchmark
+  def replayMvcEppMT(state: MvcEppBenchState): Unit = {
+    implicit val scheduler: Scheduler = monix.execution.Scheduler.Implicits.global
+    val runTask                       = prepareRunTask(state)
+    processErrors(Await.result(runTask.runAsync, Duration.Inf))
+  }
+}
+
+object EvalBench {
+
+  @State(Scope.Benchmark)
+  class MvcEppBenchState extends BenchStateBase {
+    val singleThreadedScheduler: Scheduler = TrampolineScheduler.apply(
+      Scheduler.singleThread(name = "mvcepp-1"),
+      ExecutionModel.SynchronousExecution)
+
+    override val rhoScriptSource: String = "/rholang/mvcepp.rho"
+  }
+
+  trait BenchStateBase {
+    private val dbDir: Path   = Files.createTempDirectory("rchain-storage-test-")
+    private val mapSize: Long = 1024 * 1024 * 1024
+
+    val rhoScriptSource: String
```

Comment:
Could we have this be:
```scala
val rholangSourceFile: Path
```

(see how I use the `Path` in other comment)

Reply:
Actually that's not a path, but resource key. And function getResourceAsStream doesn't have Path overload. However when a file located in resources it automatically gets a key equal to it's relative path (so it's a path, lol). To avoid cases like "Path constructor 'fixed' incoming value, for example by replacing '/' with '\'" on Windows I decided to leave type String,

Reply:
I meant to delete this comment before submitting my review, as I realized yr approach makes sense given the fact that you are including the rholang source file as a resource. 

So this is fine.